service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/gceinstance/CreateGceInstanceStep.java

```diff
@@ -173,14 +176,14 @@ private static Instance createInstanceModel(
   static Instance setFields(
       ApiGcpGceInstanceCreationParameters creationParameters,
       String instanceId,
+      String projectId,
       String zone,
       String serviceAccountEmail,
       String workspaceUserFacingId,
       String cliServer,
       Instance instance,
       String gitHash) {
-    String machineType =
-        String.format("zones/%s/machineTypes/%s", zone, creationParameters.getMachineType());
+    String machineType = GcpUtils.toMachineTypeString(zone, creationParameters.getMachineType());
```

Comment:
If this input is bogus, what kind of error comes out?

Reply:
I guess it depends how bogus the input is. If it's just some nonsense it will probably error with `Invalid value for field 'resource.machineType': 'stuff'. The URL is malformed` but if it's just not supported for the zone you get `Invalid value for field 'resource.machineType': 'stuff. Machine type with name 'stuff' does not exist in zone 'zone'`